src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -16,7 +17,7 @@ namespace Npgsql
     /// A buffer used by Npgsql to read data from the socket efficiently.
     /// Provides methods which decode different values types and tracks the current position.
     /// </summary>
-    public sealed partial class NpgsqlReadBuffer
+    public sealed partial class NpgsqlReadBuffer : IDisposable
```

Comment:
I think this is not a real issue for a quasi-internal API, so I'm going to resolve this.

Reply:
NpgsqlReadBuffer is indeed not a "real" public API - it's public only because type handlers can be written outside of Npgsql, and these need to access NpgsqlReadBuffer.

FWIW I think we should consider moving the type handler APIs into an Internal namespace or otherwise clearly document that these APIs aren't subject to the same API stability guarantees...